### PR TITLE
Add R language server

### PR DESCRIPTION
**Pull request by phelipetls**
Add support for the R language server.

https://asciinema.org/a/TxCYrJKHzib6i9cygp0jqexlK

### CODE REVIEW

File: lua/nvim_lsp/r_language_server.lua [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    ]];
+    default_config = {
+      root_dir = [[root_pattern(".git") or os_homedir]]
+    };
+  };
+}
+
+
+local install = function()
+  local f = io.popen([[Rscript -e "install.packages('languageserver')"]])
```
Comment (h-michael) [inline lua/nvim_lsp/r_language_server.lua:33]: If this is the first time to install R package, this script always failed.

```sh
~> Rscript -e "install.packages('languageserver')"
Warning in install.packages("languageserver") :
  'lib = "/usr/lib/R/library"' is not writable
Error in install.packages("languageserver") : unable to install packages
Execution halted
```

```sh
~> R

R version 4.0.0 (2020-04-24) -- "Arbor Day"
Copyright (C) 2020 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> install.packages('languageserver')
Warning in install.packages("languageserver") :
  'lib = "/usr/lib/R/library"' is not writable
Would you like to use a personal library instead? (yes/No/cancel)
```

File: lua/nvim_lsp/r_language_server.lua [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+configs.r_language_server = {
+  default_config = {
+    cmd = {"R", "--slave", "-e", "languageserver::run()"};
+    filetypes = {"r", "rmd"};
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or vim.loop.os_homedir()
+    end;
+    log_level = vim.lsp.protocol.MessageType.Warning;
+  };
+  docs = {
```
Comment (h-michael) [inline lua/nvim_lsp/r_language_server.lua:13]:
```suggestion
  docs = {
  package_json = "https://raw.githubusercontent.com/REditorSupport/vscode-r-lsp/master/package.json";
```

**Comment by h-michael** [comment]
@phelipetls 

`:LspInstall r_language_server` result is as follows:

```
Warning in install.packages("languageserver", lib = Sys.getenv("R_LIBS_USER")) :
'lib = "~/R/x86_64-pc-linux-gnu-library/4.0"' is not writable
Error in install.packages("languageserver", lib = Sys.getenv("R_LIBS_USER")) :
unable to install packages
```

Is it the expected result?

**Comment by phelipetls** [comment]
It isn't, sorry about that! I thought the function would create the directory, but maybe it doesn't. I'll work on it later.

**Comment by h-michael** [comment]
@phelipetls Should I set `$R_LIBS_USER`?

**Comment by phelipetls** [comment]
> @phelipetls Should I set `$R_LIBS_USER`?

I think R sets it automatically. On my machine, it's only visible when I call this function. Not sure if just because I already have a local site-packages.

The documentation led me to believe that this variable is set automatically to a convenient directory depending on the operating system.

> If lib is omitted or is of length one and is not a (group) writable directory, in interactive use the code offers to create a personal library tree (the first element of Sys.getenv("R_LIBS_USER")) and install there

File: lua/nvim_lsp/r_language_server.lua [hunk 3]
```diff
@@ -0,0 +1,71 @@
+local util = require 'nvim_lsp/util'
+local configs = require 'nvim_lsp/configs'
+
+configs.r_language_server = {
+  default_config = {
+    cmd = {"R", "--slave", "-e", "languageserver::run()"};
```
Comment (h-michael) [inline lua/nvim_lsp/r_language_server.lua:6]:
```suggestion
    cmd = {"R", "--slave", "-e", "'languageserver::run()'"};
```

Comment (phelipetls) [reply lua/nvim_lsp/r_language_server.lua:6]: It doesn't work here when I do this, so I left as it is.

**Comment by h-michael** [comment]
@phelipetls Thanks a lot :)

**Pull request merged**